src/App.tsx

```diff
@@ -115,7 +115,7 @@ export default class App extends Component<{}, { chosenNode: TreeNode<bleble>, v
                 />
 
                 <label>Selected node: {this.state.chosenNode.data.name} : {this.state.chosenNode.id}</label>
-                <button onClick = { e => this.remove(e) }>Usuń</button>
+                <button onClick = { e => this.remove(e) } disabled = {this.state.chosenNode === this.state.ourTree ? true : false}>Usuń</button>
```

Comment:
I'm not really fond of the abstraction leakage here `this.state.chosenNode === this.state.ourTree` => does `isRoot` method sounds reasonable? Ofc then we would need an underlying mechanism for a node to know that's it's a root.

Reply:
`isRoot(nodeId)`